c/secp256k1_ripemd160_sha256_sighash_all.c

```diff
@@ -115,65 +133,48 @@ int main(int argc, char* argv[]) {
     if (ret != CKB_SUCCESS) {
       return ERROR_SYSCALL;
     }
-    if (len > WITNESS_SIZE) {
-      return ERROR_WITNESS_TOO_LONG;
-    }
-
-    witness_pos.ptr = (const uint8_t*)witness;
-    witness_pos.size = len;
-
-    arg_res = mol_cut(&witness_pos, MOL_Witness(1));
-    if (arg_res.code != 0) {
-      if (arg_res.attr != 2) {
-        return ERROR_WRONG_NUMBER_OF_ARGUMENTS;
-      } else {
-        return ERROR_ENCODING;
-      }
-    } else if (arg_res.attr != 2) {
-      return ERROR_WRONG_NUMBER_OF_ARGUMENTS;
-    }
-
-    bytes_res = mol_cut_bytes(&arg_res.pos);
-    if (bytes_res.code != 0) {
-      return ERROR_ENCODING;
+    if (len != RECOVERABLE_SIGNATURE_SIZE + NONE_COMPRESSED_PUBKEY_SIZE
```

Comment:
Don't need to support recoverable signature.

Reply:
The previous version supports recoverable signature, and there are some unit tests testing this feature.